torch/distributed/distributed_c10d.py

```diff
@@ -1465,6 +1466,126 @@ def reduce_scatter(output,
         work.wait()
 
 
+def all_to_all(output,
+               input,
+               output_split_sizes=[],
+               input_split_sizes=[],
+               group=group.WORLD,
+               async_op=False):
+    """
+    Each process splits input tensor equally and then scatters the split list to all processes in a group.
+    Then concatenate the received tensors from all the processes in the group and return single output tensor.
+
+    Arguments:
+        output (Tensor or list[Tensor]): Gathered cancatenated output tensor if single Tensor is given
+            otherwise list of tensors to be gathered one per rank
+        input (Tensor or list[Tensor]): Input tensor to scatter if single tensor is given, otherwise
+            list of tensors to scatter one per rank.
+            Either both `input` and `output` should be list of tensors or single tensor but not a mix
+        output_split_sizes: (list[Int], optional): Output split sizes for dim 0 when `output` is single tensor,
+            if specified None or empty, dim 0 of `output` tensor must divide equally by `world_size`. Ignored if 
+            `output` is a list of tensor
+        input_split_sizes: (list[Int], optional): Output split sizes for dim 0 when `input` is single tensor,
+            if specified None or empty, dim 0 of `input` tensor must divide equally by `world_size`. Ignored if 
+            `input` is a list of tensor
+        group (ProcessGroup, optional): The process group to work on
+        async_op (bool, optional): Whether this op should be an async op
+
+    Returns:
+        Async work handle, if async_op is set to True.
+        None, if not async_op or if not part of the group.
+
+    Examples::
+        >>> input = torch.arange(4) + rank * 4
+        >>> input
+        tensor([0, 1, 2, 3])     # Rank 0
```

Comment:
Could you please do the following to generate html docs and check if the docs here are rendered properly?

```
cd docs/
pip install -r requirements.txt
make html
```

You can check the result by using this [gist](https://gist.github.com/colesbury/df218a96334dbd31ee0e93760f70f114) to serve your docs and paste a screenshot in this PR.

Reply:
Here are the screen shots...
![image](https://user-images.githubusercontent.com/8791375/76064388-8eec7c80-5faf-11ea-916d-52001eb4bcf5.png)
![image](https://user-images.githubusercontent.com/8791375/76064508-d1ae5480-5faf-11ea-9d61-e3be74a0fa00.png)
![image](https://user-images.githubusercontent.com/8791375/76064555-e7237e80-5faf-11ea-8a1e-e526cdce9dc1.png)